packages/multichain-account-service/src/providers/SolAccountProvider.ts

```diff
@@ -15,4 +17,52 @@ export class SolAccountProvider extends BaseAccountProvider {
       account.metadata.keyring.type === (KeyringTypes.snap as string)
     );
   }
+
+  async #createAccount(opts: {
+    entropySource: EntropySourceId;
+    derivationPath: `m/${string}`;
+  }) {
+    // NOTE: We're not supposed to make the keyring instance escape `withKeyring` but
+    // we have to use the `SnapKeyring` instance to be able to create Solana account
+    // without triggering UI confirmation.
+    // Also, creating account that way won't invalidate the Snap keyring state. The
+    // account will get created and persisted properly with the Snap account creation
+    // flow "asynchronously" (with `notify:accountCreated`).
+    const createAccount = await this.withKeyring<
+      SnapKeyring,
+      SnapKeyring['createAccount']
+    >({ type: KeyringTypes.snap }, async ({ keyring }) =>
+      keyring.createAccount.bind(keyring),
+    );
+
+    // Create account without any confirmation nor selecting it.
+    const keyringAccount = await createAccount(
+      SolAccountProvider.SOLANA_SNAP_ID,
+      opts,
+      {
+        displayAccountNameSuggestion: false,
+        displayConfirmation: false,
+        setSelectedAccount: false,
+      },
+    );
```

Comment:
Do you think we should move this class outside the package? I see that we could just add a flag to manage the capabilities. After thinking about it a bit more, ideally these classes would be part of each snap monorepo but that could limit our ownership over them and create some management complexity around multichain accounts

Reply:
➕ to the keyring refactor point